File: Railflow.Core/ValueObjects/Connection.cs
using Railflow.Core.Entities;

namespace Railflow.Core.ValueObjects;

public record Connection(IEnumerable<SubConnection> SubConnections, Guid StartStopId, string StartStationName,
    Guid EndStopId, string EndStationName, TimeOnly StartHour, TimeOnly EndHour, long Price);

File: client-app/src/app/models/connection.ts
import { Schedule } from "./schedule";
import { Stop } from "./stop";

export interface SubConnection {
  schedule: Schedule;
  stops: Stop[];
}

export interface Connection {
  subConnections: SubConnection[];
  startStopId: string;
  startStationName: string;
  endStopId: string;
  endStationName: string;
  startHour: string;
  endHour: string;
  price: number
}